### DISCUSSION TITLE
Ripple Effect Area Mask

### DISCUSSION THREAD

**Discussion by DasLixou** [AvaloniaUI/Avalonia#13314] [Q&A]
I'm making a ripple effect for a rounded button which is an Ellipse. Now the problem is that it clamps at the raw rectangle of its parent but not at the rounded border, so it overflows in an ugly way:
![image](https://github.com/AvaloniaUI/Avalonia/assets/82600264/731baa85-99e5-4a89-b836-050f6244b0f5)

I even tried sniffing around in the low level code but I couldn't find any way to make it work.
This is an essential feature!

**Comment by timunie** [comment]
please check Material.Avalonia, I believe they ship such an effect already.

**Reply by DasLixou** [reply]
Yeah they have the problem. I tried to fix it in their code but couldn't.

**Reply by DasLixou** [reply]
@timunie any other suggestion?

**Reply by DasLixou** [reply]
bump
